### PR TITLE
Refactor & design

### PR DESCRIPTION
**Pull request by luqmaan**
Refactor main.js and add some more stylings.

You can see what this looks like at luqmaan.com/open-data-progress-report.

Its also usable on mobile.

image

We can probably port these changes into the alt layout without too much difficulty.

I think the alt layout is better, I just saw it too late after starting this.

### CODE REVIEW

**Comment by haileypate** [comment]
Wow, love this so far as I'm seeing it on my iPhone! I especially like how it tells me how long ago the dataset was uploaded. If that was there before... my humble apologies for not noticing sooner ;)

Mateo, I defer executive decisions on pull requests to you. Luqmaan, dang you design too?! I feel like this style gets us closer to 'gelling' with both Open Austin's brand and COA's current site theme. Yay!

**Comment by luqmaan** [comment]
Heh, nvm the works on mobile part. Looks horrible on my phone.

**Comment by haileypate** [comment]
Hrm. Well maybe it just looks good on my big giant iPhone 6plus :D

File: index.html [hunk 1]
```diff
@@ -16,22 +16,27 @@
         <script src="js/vendor/modernizr-2.8.3.min.js"></script>
     </head>
     <body>
-        <div class="tab">
-            <a href="/open-data-progress-report/alt-version.html">Click here to see an alternative layout.</a>
-        </div>
```
Comment (mateoclarke) [inline index.html:21]: So I know this was ugly but it served a purpose to compare the two layouts and the feedback we got was in favor of the alternative "stacked layout". We either need to keep this or we should really just go ahead and adopt the columns and rows layout.

File: index.html [hunk 2]
```diff
@@ -16,22 +16,27 @@
...
-            <b>2)</b> conduct a <b>data inventory</b>, 
-            <b>3)</b> submit an <b>open data participation plan</b>, and
-            <b>4)</b> <b>publish at least three new datasets</b> to the City's open data portal.
-            Departments continue to work with CTM to meet these goals, and we will update this dataset regularly to reflect their progress untifurther notice. <i>This progress report evaluates <span class="dept-count">xx</span> CoA departments.</i>
-            </p>
-
-            <p><a href="https://data.austintexas.gov/resource/8qty-vat2">Source data</a> updated: <span class="updated-date"></span></p></i>
-
+            <div class="content">
+                <a class="open-atx-logo" href="https://open-austin.org">
```
Comment (mateoclarke) [inline index.html:21]: Yes!

File: index.html [hunk 3]
```diff
@@ -53,20 +58,62 @@ <h2 class="symbols-title">Compliance Categories</h2>
...
+            <h2 class="secondary-title">Department Progress</h2>
 
+            <script id="metadata-template" type="text/x-handlebars-template">
+                <p><a href="https://data.austintexas.gov/resource/8qty-vat2" target="_blank">Source data</a> last updated {{updated}}</p></i>
+            </script>
+        </div>
 
+        <div class="report-wrapper"><div class="report-container"></div></div>
 
+        <script id="card-template" type="text/x-handlebars-template">
```
Comment (mateoclarke) [inline index.html:73]: fancy! :nail_care: 
Handlebars template is a way nicer option than my hacky HTML block variable template inside main.js. I started the original one with @sharlalikesyou so I wanted to keep it simple and minimize outside libraries, but this makes the best sense.

File: js/main.js [hunk 4]
```diff
@@ -1,123 +1,73 @@
...
-      var imgFilename = data[i].dept.toLowerCase().replace(/\s+/g, "-");
-      var imgPath = "img/" + imgFilename + ".svg";
-
-      $(".card-" + i + " .dept-icon").attr("src", imgPath)
 
+function renderReportMetadata(data) {
+    var source = $('#metadata-template').html();
+    var template = Handlebars.compile(source);
+    var context = {
+        updated: moment.unix(data.rowsUpdatedAt).fromNow()
```
Comment (mateoclarke) [inline js/main.js:26]: Good call on moment.js. Time since update is a nice touch instead of just the date.

File: js/main.js [hunk 5]
```diff
@@ -1,123 +1,73 @@
...
+        else if (['in progress'].indexOf(status) !== -1) {
+            return 'yellow';
+        }
+        else if (['not started', 'not yet', 'unknown'].indexOf(status) !== -1) {
+            return 'red';
+        }
+        else {
+            return 'gray';
+        }
+    }
```
Comment (mateoclarke) [inline js/main.js:46]: Way nicer than my endless if elses. When pairing with a code newbie, I wanted to keep things simple, but this reads so much better and is actually more simple itself.

**Comment by mateoclarke** [comment]
@luqmaan, Thanks for adding some :green_heart:.

I think we've decided to roll with the [stacked layout](http://open-austin.github.io/open-data-progress-report/alt-version.html). Did you check that one out? Would it be too much to ask for you to update the Handlebars template and make any necessary changes to `alt.js`? 

Or... I can go ahead and merge these changes for `index.html` and `main.js` and learn from what you did here to implement those changes to `alt-version.html` and `alt.js`. 

You're call...

**Comment by luqmaan** [comment]
You should learn!

**Comment by mateoclarke** [comment]
:+1:

**Comment by mateoclarke** [comment]
cool if I hold off on deploying changes to `master` until then?

**Comment by luqmaan** [comment]
Si

**Pull request merged**